src/Classes/Utils/CanvasUtils.ts

```diff
@@ -0,0 +1,348 @@
+
+export class CanvasUtils {
+    public static paintBase(context: CanvasRenderingContext2D, dimension: IDimension, baseColor: string = "rgba(255, 255, 255, 0)"): void {
+        context.fillStyle = baseColor;
+        context.fillRect(0, 0, dimension.width, dimension.height);
+    }
+
+    public static clear(context: CanvasRenderingContext2D, dimension: IDimension) {
+        context.clearRect(0, 0, dimension.width, dimension.height);
+    }
+
+    public static drawPolygonMask(context: CanvasRenderingContext2D, rawData: ICoordinates[], strokeStyle: string, lineWidth: number) {
+        context.beginPath();
+        context.moveTo(rawData[0].x, rawData[0].y);
+
+        for (let i = 1; i < rawData.length; i++) {
+            context.lineTo(rawData[i].x, rawData[i].y);
+        }
+
+        context.closePath();
+        context.strokeStyle = strokeStyle;
+        context.lineWidth = lineWidth;
+        context.stroke();
+    }
+
+    public static drawLineLinked(context: CanvasRenderingContext2D, width: number, begin: ICoordinates, end: ICoordinates, backgroundMask: boolean, colorLine: IRgb, opacity: number) {
+        context.save();
+
+        if (backgroundMask) {
+            context.globalCompositeOperation = 'destination-out';
+        }
+
+        if (colorLine) {
+            context.strokeStyle = `rgba(${colorLine.r},${colorLine.g},${colorLine.b},${opacity})`;
+        }
+
+        context.lineWidth = width;
+        // this.ctx.lineCap = "round"; /* performance issue */
+        /* path */
+        context.beginPath();
+        context.moveTo(begin.x, begin.y);
+        context.lineTo(end.x, end.y);
+        context.stroke();
+        context.closePath();
+        context.restore();
+    }
+
+    public static drawConnectLine(context: CanvasRenderingContext2D, width: number, lineStyle: CanvasGradient, begin: ICoordinates, end: ICoordinates) {
+        context.beginPath();
+        context.lineWidth = width;
+        context.strokeStyle = lineStyle;
+        context.moveTo(begin.x, begin.y);
+        context.lineTo(end.x, end.y);
+        context.stroke();
+        context.closePath();
+    }
+
+    public static gradient(context: CanvasRenderingContext2D, p1: Particle, p2: Particle, midColor: string): CanvasGradient | undefined {
+        const gradStop = Math.floor(p2.radius / p1.radius);
+
+        if (!p1.color || !p2.color) {
+            return;
+        }
+
+        const sourcePos = p1.position;
+        const destPos = p2.position;
+        const grad = context.createLinearGradient(sourcePos.x, sourcePos.y, destPos.x, destPos.y);
+
+        grad.addColorStop(0, Utils.getStyleFromColor(p1.color));
+        grad.addColorStop(gradStop > 1 ? 1 : gradStop, midColor);
+        grad.addColorStop(1, Utils.getStyleFromColor(p2.color));
+
+        return grad;
+    }
+
+    public static drawGrabLine(context: CanvasRenderingContext2D, width: number, begin: ICoordinates, end: ICoordinates, colorLine: IRgb, opacity: number) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Exceeds maximum line length of 120](https://app.codacy.com/manual/ar3s/tsparticles/pullRequest?prid=5171185)